pkg/syncer/admissionhandler/validatepvcannotation.go

```diff
@@ -0,0 +1,87 @@
+		log.Warn(reason)
+		return admission.Allowed(reason)
+	}
+
+	if request.Operation == admissionv1.Create {
+		// disallow PVC Create with AnnGuestClusterRequestedTopology key but empty value
+		annGuestClusterRequestedTopologyValue, ok := newPVC.Annotations[common.AnnGuestClusterRequestedTopology]
+		if ok && annGuestClusterRequestedTopologyValue == "" {
+			return admission.Denied(CreatePVCWithAnnGuestClusterRequestedTopologyButEmptyValue)
+		}
+	} else if request.Operation == admissionv1.Update {
+		oldPVC := corev1.PersistentVolumeClaim{}
+		if err := json.Unmarshal(request.OldObject.Raw, &oldPVC); err != nil {
+			reason := "skipped validation when failed to deserialize PVC from old request object"
+			log.Warn(reason)
+			return admission.Allowed(reason)
+		}
+
+		// disallow PVC Update on AnnGuestClusterRequestedTopology
+		oldAnnGuestClusterRequestedTopologyValue, oldOk := oldPVC.Annotations[common.AnnGuestClusterRequestedTopology]
+		newAnnGuestClusterRequestedTopologyValue, newOk := newPVC.Annotations[common.AnnGuestClusterRequestedTopology]
+		if oldOk && newOk {
+			// disallow changing AnnGuestClusterRequestedTopology to a different value
+			if oldAnnGuestClusterRequestedTopologyValue != newAnnGuestClusterRequestedTopologyValue {
+				return admission.Denied(UpdatePVCAnnotation + ", " + common.AnnGuestClusterRequestedTopology)
+			}
+		} else if oldOk {
+			// disallow removing the annotation of AnnGuestClusterRequestedTopology in an existing PVC
+			return admission.Denied(RemovePVCAnnotation + ", " + common.AnnGuestClusterRequestedTopology)
+		} else if newOk {
+			// disallow adding an annotation of AnnGuestClusterRequestedTopology in an existing PVC
+			return admission.Denied(AddPVCAnnotation + ", " + common.AnnGuestClusterRequestedTopology)
+		}
+
+		// disallow PVC Update on AnnVolumeAccessibleTopology to another non-empty value
+		oldAnnVolumeAccessibleTopologyValue, oldOk := oldPVC.Annotations[common.AnnVolumeAccessibleTopology]
+		newAnnVolumeAccessibleTopologyValue, newOk := newPVC.Annotations[common.AnnVolumeAccessibleTopology]
+		if oldOk && newOk {
+			if oldAnnVolumeAccessibleTopologyValue != newAnnVolumeAccessibleTopologyValue {
+				// disallow changing AnnGuestClusterRequestedTopology to another a different value except
+				// only allow changing AnnGuestClusterRequestedTopology from an empty value to a non-empty value
+				if !(oldAnnVolumeAccessibleTopologyValue == "" && newAnnVolumeAccessibleTopologyValue != "") {
+					return admission.Denied(UpdatePVCAnnotation + ", " + common.AnnVolumeAccessibleTopology)
+				}
+			}
+		} else if oldOk {
+			// disallow removing the annotation of AnnVolumeAccessibleTopology in an existing PVC
+			return admission.Denied(RemovePVCAnnotation + ", " + common.AnnVolumeAccessibleTopology)
+		} else if newOk {
+			// Note: allow adding an annotation of AnnGuestClusterRequestedTopology in an existing PVC for now;
```

Comment:
change AnnGuestClusterRequestedTopology -> AnnVolumeAccessibleTopology here in the comment.

Reply:
Done